JustSaying/JustSayingBus.cs

```diff
@@ -126,31 +125,12 @@ public void Start()
                 {
                     foreach (var queueSubscriber in regionSubscriber.Value)
                     {
-                        queueSubscriber.Value.Listen();
+                        queueSubscriber.Value.Listen(cancellationToken);
                     }
                 }
 
-                Listening = true;
-            }
-        }
-
-        public void Stop()
-        {
-            lock (_syncRoot)
-            {
-                if (!Listening)
-                {
-                    return;
-                }
-
-                foreach (var regionSubscriber in _subscribersByRegionAndQueue)
-                {
-                    foreach (var queueSubscriber in regionSubscriber.Value)
-                    {
-                        queueSubscriber.Value.StopListening();
-                    }
-                }
-                Listening = false;
+                cancellationToken.Register(() => _subscribersAreListening = false);

```

Comment:
@adammorr Just realised we should probably not do this, as this creates a disposable subscription and can leak (I think). Might need this to set `_subscribersAreListening` with a try-finally inside the listen loop.